### PR TITLE
Allows connect to server version 19.16.19.85

### PR DESCRIPTION
**Pull request by aastafiev**
Hello,

This PR brings backward compatibility with ClickHouse version 19.16.19.85.

This query won't work because readonly field doesn't exit.
`SELECT name, value, readonly FROM system.settings LIMIT 10000`
Consequently, clickhouse-connect can't connect to the server.

I suggest including common setting "readonly" to be able set it in runtime and change query to 
`SELECT name, value FROM system.settings LIMIT 10000`
for server version less than 22.3.

So, client could set "readonly" setting before connecting to the server like so:
```
import clickhouse_connect
from clickhouse_connect import common

common.set_setting('readonly', 1)
client = clickhouse_connect.get_client(host='localhost', user='default', port=9123)
```

### CODE REVIEW

**Comment by genzgd** [comment]
Thanks for your submission!  A couple of general comments --

- If possible, please start your branch from a more recent version so there aren't so many merge commits
- In your IDE, please set your line wrapping to 120 characters so not so many existing lines get reformatted

As for your main change, it seems fine to allow backward compatibility.  I think you could simplify the code quite a bit though by keeping just one query for system.settings, something like this:

```python
if self.min_version('22.3'):   # We should check this version, when was the readonly flag introduced?
     read_only = 'readonly'
else:
     read_only = common.get_setting('readonly')
server_settings = self.query(f'SELECT name, value, {readonly} as readonly FROM system.settings LIMIT 10000')
```

File: playtest.py [hunk 1]
```diff
@@ -1,19 +1,37 @@
+from io import BytesIO
```
Comment (genzgd) [inline playtest.py:1]: Could you add a separate test module for your testing?  This file is just a small example of how to connect to play.clickhouse.com

File: clickhouse_connect/common.py [hunk 2]
```diff
@@ -59,3 +60,4 @@ def _init_common(name: str, options: Sequence[Any], default: Any):
 _init_common('dict_parameter_format', ('json', 'map'), 'json')
 _init_common('invalid_setting_action', ('send', 'drop', 'error'), 'error')
 _init_common('product_name', (), '')
+_init_common('readonly', (0, 1, 2), 0)
```
Comment (genzgd) [inline clickhouse_connect/common.py:63]: I think we could limit this to 0 and 1 since it's only necessary for older versions

File: .gitignore [hunk 3]
```diff
@@ -15,10 +16,13 @@
...
+.env
 
 # Python cruft
 *.pyc
 .python-version
 
 # C extensions
 *.pyd
+
+playtest_custom.py
```
Comment (genzgd) [inline .gitignore:28]: You can put your own test files under the `dev` directory without a special case in .gitignore

**Review by genzgd — CHANGES_REQUESTED** [review]

File: playtest.py (hunk 1 above)
Comment (aastafiev) [reply playtest.py:1]: fixed tests for servers version started from 19.16.19.85

File: clickhouse_connect/common.py (hunk 2 above)
Comment (aastafiev) [reply clickhouse_connect/common.py]: done

File: .gitignore (hunk 3 above)
Comment (aastafiev) [reply .gitignore]: done

**Comment by aastafiev** [comment]
Hello,

Thank you for your feedback.
I've taken into account your commets and passed through all tests starting server version 19.16.19.85.

File: playtest.py (hunk 1 above)
Comment (aastafiev) [reply playtest.py]: revert back original file

**Comment by genzgd** [comment]
Thanks for backporting all those tests!  That's awesome.  We can't merge this until you sign the CLA though.  Also if you could remove the flake8 comments (I see a couple of `# noqa comments`) that would reduce the noise a bit.

**Comment by aastafiev** [comment]
Will do it asap. 
I don’t understand how to sign CLA. Could you give me instructions on it?


---------------------
Sincerely yours,
Aleksey Astafiev

**Comment by aastafiev** [comment]
I've signed the CLA but for some reason aastafiev marked as signed but Aleksey Astafyev not. Is it critical?

**Review by genzgd — APPROVED** [review]

**Comment by aastafiev** [comment]
Thank you for acceptence.
Could you reveal when to expect a new release with this feature?

**Comment by genzgd** [comment]
Is it something you need available for install soon?  We can do a release if you need it, but otherwise I would probably wait until there's a new feature or important bug fix before cutting another release, since I can't imagine many folks are running ClickHouse 19.x :)

**Comment by aastafiev** [comment]
If it wouldn't take more than month I could wait :)

**Comment by genzgd** [comment]
It's never that long :).   If there's not anything else to add in the next week or so I'll cut a release.

**Pull request merged**